cmd/gke-gcloud-auth-plugin/cred/cred.go

```diff
@@ -80,6 +81,9 @@ func newPlugin(tokenProvider tokenProvider) *plugin {
 // Options struct inputs to PrintCred
 type Options struct {
 	UseApplicationDefaultCredentials bool
+	UseEdgeCloud                     bool
```

Comment:
Is it maybe better to make this an enum or something? I can imagine a future world where someone else wants to add a new provider, and somehow multiple provider flags get specified. This should obviously error out, but ideally it should happen at flag parsing and processing time, not deep in the code.

It would also allow switching the Edge Cloud/ADC check from a set of ifs to a switch, which might be nicer if more providers are added.

Reply:
Added enum